Add boilerplate verification scripts

Hii-Himanshu opened the pull request:
This PR adds the initial boilerplate verification setup for #19:

- Created the `hack/boilerplate/` folder and moved `boilerplate.go.txt` into it.
- Added a copy of `hack/verify-boilerplate.sh` based on the Kubernetes script.
- Added `hack/verify-all.sh`, which calls `verify-boilerplate.sh`.

please let me know if anything needs to be adjusted.

Thank you @Priyankasaggu11929 for your help, I am new to contributing and happy to learn more.

Karthik-K-N commented on hack/verify-boilerplate.sh:
```diff
@@ -0,0 +1,41 @@
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+boilerDir="${KUBE_ROOT}/hack/boilerplate"
+boiler="${boilerDir}/boilerplate.py"
```
boilerplate.py is missing this PR

Hii-Himanshu commented:
Thanks for the review! I have added `boilerplate.py` from the Kubernetes repo and fixed the newline issue.  
Please let me know if anything else is needed.

Karthik-K-N commented on hack/boilerplate/boilerplate.py:
```diff
@@ -0,0 +1,244 @@
+
+
+if __name__ == "__main__":
+    sys.exit(main())
```
Add the new line.

Karthik-K-N commented on hack/verify-all.sh:
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
```
I feel this file is not needed as of now, as we have only one verify to run. Just update the Makefile to add
```
.PHONY: verify-boilerplate
verify-boilerplate: ## Verify boilerplate text exists in each file
	 ./hack/verify-boilerplate.sh
```
And validate that it works by running
`make verify-boilerplate`

Karthik-K-N commented on hack/boilerplate/boilerplate.py:
```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python
```
```suggestion
#!/usr/bin/env python3
```

ajaysundark commented:
/unassign @Priyankasaggu11929

Priyankasaggu11929 commented:
Thanks for the changes @Hii-Himanshu.

Could you please address the following 2 things as well -

- request in this comment https://github.com/kubernetes-sigs/node-readiness-controller/pull/21#discussion_r2588013487

- squash the commits into a few (maybe 2 - one for Makefile change, and one for the rest of the changes)

Priyankasaggu11929 commented on Makefile:
```diff
@@ -147,6 +147,10 @@ lint-api-fix: $(GOLANGCI_LINT_KAL)
+.PHONY: verify-boilerplate
+verify-boilerplate: ## Verify boilerplate text exists in each file
+	 ./hack/verify-boilerplate.sh
+
```
lets update this to now call the `verify-all.sh` script (which will in turn take care of calling `verify-boilerplate.sh`)

Priyankasaggu11929 commented on Makefile:
```diff
@@ -147,6 +147,10 @@ lint-api-fix: $(GOLANGCI_LINT_KAL)
 
+.PHONY: verify-boilerplate
+verify-boilerplate: ## Verify boilerplate text exists in each file
+	 ./hack/verify-all.sh
```
>  ran make verify target, it returns:
> 
> make: *** No rule to make target `verify'. Stop.
> 
> I instead ran the verification script directly:
> ./hack/verify-all.sh
> 
> It produced no output (I think which indicates success).
> Please let me know if anything else is needed.

That's because this Makefile target needs updating to 


```suggestion
.PHONY: verify
verify:
	 ./hack/verify-all.sh
```

Hii-Himanshu commented:
Hi @Priyankasaggu11929 ,
I’ve pushed the requested updates:

- Added the missing boilerplate files (.sh, Dockerfile, Makefile)
- Added boilerplate_test.py
- Updated the Makefile and confirmed that make help now shows the verify target

When running the verification the first time:
make verify
./hack/verify-all.sh

I saw an error about a missing boilerplate header:

Boilerplate header is wrong for: .devcontainer/post-install.sh
make: *** [verify] Error 1

I added the required header to that file, and after rerunning:
make verify
there was no additional output. Everything seems to be passing now.

Please let me know if anything else needs to be updated!
